command/server/config.go

```diff
@@ -464,13 +464,18 @@ func CheckConfig(c *Config, e error) (*Config, error) {
 		return c, e
 	}
 
-	if len(c.Seals) == 2 {
+	switch len(c.Seals) {
+	case 2:
 		switch {
 		case c.Seals[0].Disabled && c.Seals[1].Disabled:
 			return nil, errors.New("seals: two seals provided but both are disabled")
 		case !c.Seals[0].Disabled && !c.Seals[1].Disabled:
 			return nil, errors.New("seals: two seals provided but neither is disabled")
 		}
+	case 1:
+		if c.Seals[0].Disabled && c.Seals[0].Recover {
```

Comment:
Might be worth adding a note why this isn't an issue for the two-seal case (I believe, since the recovery seal isn't necessary for operation, it could be disabled but the other (perhaps auto-unseal) one is enabled).

Reply:
Related: do we intend for recovery seal to be used in conjunction with seal migration?

Reply:
I think the answer is no, but it would be nice if that could be supported, e.g., I've lost my aws key, please help me migrate away from it using the recovery keys.

Reply:
Ok, on closer review it looks like the answer is yes, but then I'm not sure this is right.  I would say we shouldn't allow for a single seal to be disabled, and that in a migration (two seal) scenario, it doesn't make sense to have the non-disabled seal (the migration target) have recovery keys enabled.

Reply:
I was not envisioning allowing migration while doing recovery unseal, but you make a good point.  I should at least guard against recovery being enabled on the target seal.